include/boost/core/empty_value.hpp

```diff
@@ -95,9 +95,45 @@ class empty_value {
 };
 
 #if !defined(BOOST_NO_TEMPLATE_PARTIAL_SPECIALIZATION)
+template<class T>
+class empty_value_base
```

Comment:
This class should be in `namespace detail`. And it should only be used `#if defined(BOOST_MSVC)`, otherwise the previous implementation should be in effect.